zebrad/src/components/tokio.rs

```diff
@@ -34,6 +34,7 @@ impl TokioComponent {
 /// Zebrad's graceful shutdown function, blocks until one of the supported
 /// shutdown signals is received.
 async fn shutdown() {
+    zebra_network::IS_SHUTTING_DOWN.store(true, Ordering::Relaxed);
     imp::shutdown().await;
```

Comment:
This code sets `IS_SHUTTING_DOWN` to true, then waits for shutdown.

Instead, set `IS_SHUTTING_DOWN` to true after:
```rust
        signal(kind)
            .expect("Failed to register signal handler")
            .recv()
            .await;
```
and
```rust
        tokio::signal::ctrl_c()
            .await
            .expect("listening for ctrl-c signal should never fail");
```

You might also want to add a small unit test that checks that `IS_SHUTTING_DOWN` is false:
* at the start of the test, and
* after you've spawned a blocking `shutdown()` task

We don't need to check that it becomes true - we'll notice the panics if that part of the code ever breaks.